internal/templates/templates_testutils.go

```diff
@@ -0,0 +1,71 @@
+package templates
+
+import (
+	"sort"
+
+	"github.com/stretchr/testify/suite"
+	"golang.stackrox.io/kube-linter/internal/check"
+	"golang.stackrox.io/kube-linter/internal/diagnostic"
+	"golang.stackrox.io/kube-linter/internal/lintcontext"
+)
+
+// TemplateTestSuite is a basic testing suite for all templates
+// test with some generic helper functions
+type TemplateTestSuite struct {
+	suite.Suite
+
+	Template check.Template
+}
+
+// TestCase represents a single test case which can be verified under a LintContext
+type TestCase struct {
+	Param       interface{}
+	Diagnostics []diagnostic.Diagnostic
+	ExpectError bool
```

Comment:
Name this `ExpectInstantiationError` or so, to make it clear this is an error during instantiation?